### PR TITLE
Rough first pass of paper-toast

### PR DESCRIPTION
**Pull request by garlicnation**
Missing many features from 0.5 version. I want to button up the style before porting things over.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -2,9 +2,11 @@
   "name": "paper-toast",
   "private": true,
   "dependencies": {
-    "core-media-query": "Polymer/core-media-query#master",
-    "core-overlay": "Polymer/core-overlay#master",
+    "polymer": "Polymer/polymer#0.8.0-rc.4",
```
Comment (cdata) [inline bower.json]: Update for `rc.6` :bullettrain_front:

File: bower.json [hunk 2]
```diff
@@ -2,9 +2,11 @@
...
     "font-roboto": "Polymer/font-roboto#master",
```
Comment (cdata) [inline bower.json]: We've replaced usage of this repo with `polymerelements/paper-styles`. You can pull in one of the typography imports to get the `roboto` font.

File: bower.json [hunk 3]
```diff
@@ -2,9 +2,11 @@
...
-    "paper-button": "Polymer/paper-button#master"
+    "paper-button": "Polymer/paper-button#0.8-preview"
+  },
+  "resolutions": {
+    "polymer": "0.8-preview"
```
Comment (cdata) [inline bower.json]: This should probably be changed to reflect the depended-upon `rc` tag.

File: demo.html [hunk 4]
```diff
@@ -11,23 +11,23 @@
...
   <meta name="mobile-web-app-capable" content="yes">
   <meta name="apple-mobile-web-app-capable" content="yes">
-  
-  <script src="../webcomponentsjs/webcomponents.js"></script>
-  
+
+  <script src="../webcomponentsjs/webcomponents-lite.js"></script>
+
   <link rel="import" href="paper-toast.html">
   <link rel="import" href="../font-roboto/roboto.html" >
```
Comment (cdata) [inline demo.html]: Noted re: https://github.com/Polymer/paper-toast/pull/22/files#r28729964

File: demo.html [hunk 5]
```diff
@@ -35,29 +35,29 @@
...
-<body unresolved>
+<body>
 
   <paper-button raised onclick="document.querySelector('#toast1').show()">Discard Draft</paper-button>
-  
+
   <paper-button raised onclick="document.querySelector('#toast2').show()">Get Messages</paper-button>
-  
-  <paper-button raised onclick="document.querySelector('#toast3').show()">Send Message</paper-button>
+<!--
```
Comment (cdata) [inline demo.html]: Are these demos no longer relevant?

File: index.html [hunk 6]
```diff
@@ -10,13 +10,16 @@
 <html>
 <head>
 
-  <script src="../webcomponentsjs/webcomponents.js"></script>
-  <link rel="import" href="../core-component-page/core-component-page.html">
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+  <script src="../webcomponentsjs/webcomponents-lite.js"></script>
+  <link rel="import" href="../iron-doc-viewer/iron-doc-viewer.html">
```
Comment (cdata) [inline index.html]: Let's add `polymerelements/iron-doc-viewer#^0.8.0` to the bower `devDependencies`.

File: demo.html [hunk 7]
```diff
@@ -11,23 +11,23 @@
...
-  
+
   <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1, user-scalable=yes">
   <meta name="mobile-web-app-capable" content="yes">
   <meta name="apple-mobile-web-app-capable" content="yes">
-  
-  <script src="../webcomponentsjs/webcomponents.js"></script>
-  
+
+  <script src="../webcomponentsjs/webcomponents-lite.js"></script>
```
Comment (cdata) [inline demo.html]: Let's add `webcomponents/webcomponentsjs` to the bower `devDependencies`.

File: paper-toast.html [hunk 8]
```diff
@@ -6,291 +6,138 @@
...
+        value: ""
+      }
+    },
+
+    /**
+     * Hide the toast
+     * @return {[type]} [description]
+     */
+    hide: function() {
+      console.log("hiding");
```
Comment (cdata) [inline paper-toast.html]: Lingering `console.log`.

File: paper-toast.html [hunk 9]
```diff
@@ -6,291 +6,138 @@
...
+      this.opened = false;
+    },
+
+    _toggleClasses: function(visible) {
+      this.classList.toggle('paper-toast-closed', !visible);
+      this.classList.toggle('paper-toast-open', visible);
+    },
+
+    openChanged: function() {
+      console.log("openChanged");
```
Comment (cdata) [inline paper-toast.html]: Another lingering `console.log`.

File: paper-toast.html [hunk 10]
```diff
@@ -6,291 +6,138 @@
...
+</dom-module>
+<script>
+(function() {
 
-    <div class="toast-text" flex>{{text}}</div>
-    
-    <div class="toast-text toast-action" on-tap="{{dismiss}}">
-      <content></content>
-    </div>
+  var currentToast;
```
Comment (cdata) [inline paper-toast.html]: Channeling @sjmiles for a second, let's keep this reference as a static property of the `PaperToast` constructor if possible.

File: paper-toast.html [hunk 11]
```diff
@@ -6,291 +6,138 @@
...
+      this.opened = !this.opened;
+      this.notifyPath('opened', this.opened);
+    },
+
+    /**
+     * Show the toast.
+     * @method show
+     */
+    show: function() {
+      console.log("Showing");
```
Comment (cdata) [inline paper-toast.html]: Yet another lingering `console.log`

File: paper-toast.html [hunk 12]
```diff
@@ -6,291 +6,138 @@
...
-  })();
+    },
+
+    /**
+     * Toggle the opened state of the toast.
+     * @method toggle
+     */
+    toggle: function() {
+      this.opened = !this.opened;
+      this.notifyPath('opened', this.opened);
```
Comment (cdata) [inline paper-toast.html]: Why not just set `opened` to `notify: true`?

File: paper-toast.html [hunk 13]
```diff
@@ -6,291 +6,138 @@
...
+
+    /**
+     * Show the toast.
+     * @method show
+     */
+    show: function() {
+      console.log("Showing");
+      this.opened = true;
+      if (currentToast && currentToast !== this) {
+        console.log("Hiding current toast from openChanged");
```
Comment (cdata) [inline paper-toast.html]: ƃol˙ǝlosuoɔ ɹǝɥʇouɐ ʇǝʎ llᴉʇS

File: paper-toast.html [hunk 14]
```diff
@@ -6,291 +6,138 @@
...
+        currentToast.hide();
+      }
+      currentToast = this;
+      this._toggleClasses(true);
+      console.log(this.duration);
```
Comment (cdata) [inline paper-toast.html]: :dancers:

File: index.html (hunk 6 above)
Comment (garlicnation) [inline index.html]: Done.

File: demo.html (hunk 7 above)
Comment (garlicnation) [inline demo.html]: Done.

**Comment by garlicnation** [comment]
@cdata PTAL.

File: bower.json [hunk 15]
```diff
@@ -2,9 +2,12 @@
   "name": "paper-toast",
   "private": true,
   "dependencies": {
-    "core-media-query": "Polymer/core-media-query#master",
-    "core-overlay": "Polymer/core-overlay#master",
-    "font-roboto": "Polymer/font-roboto#master",
-    "paper-button": "Polymer/paper-button#master"
+    "polymer": "Polymer/polymer#0.8.0-rc.6",
+    "paper-styles": "Polymer/font-roboto#master"
```
Comment (cdata) [inline bower.json]: I think this key's value needs to be `polymerelements/paper-styles#^0.8.0`

File: bower.json [hunk 16]
```diff
@@ -2,9 +2,12 @@
...
+  },
+  "devDependencies": {
+    "iron-doc-viewer": "polymerelements/iron-doc-viewer#^0.8.0",
+    "paper-button": "Polymer/paper-button#0.8-preview",
+    "webcomponentsjs": "webcomponents/webcomponentsjs"
```
Comment (cdata) [inline bower.json]: Can you use `#^0.6.0` for this dependency?

File: paper-toast.html [hunk 17]
```diff
@@ -6,291 +6,132 @@
...
+     * Hide the toast
+     * @return {[type]} [description]
+     */
+    hide: function() {
+      this.opened = false;
+    },
+
+    _toggleClasses: function(visible) {
+      this.classList.toggle('paper-toast-closed', !visible);
+      this.classList.toggle('paper-toast-open', visible);
```
Comment (cdata) [inline paper-toast.html]: I think I read in chat that there are Internet Explorer reasons why `Polymer.Base.toggleClass` is preferred for this. :cry:

File: paper-toast.html [hunk 18]
```diff
@@ -6,291 +6,132 @@
...
+      PaperToast.currentToast = this;
+      this._toggleClasses(true);
+      this.debounce('dismissToast', this.hide, this.duration);
+    }
 
-</script>
-</polymer-element>
+  });
+})();
+</script>
```
Comment (cdata) [inline paper-toast.html]: Missing newline @ EOF?

**Comment by cdata** [comment]
Some small details left, otherwise LGTM :+1:

**Pull request merged**